Documentation/ApiOverview/Environment/Index.rst

```diff
@@ -84,9 +88,48 @@ the web document root - a situation that is not optimal from a security point of
 .. _Environment-current-script:
 
 getCurrentScript()
-==================
+------------------
+
+Returns the path and filename to the current PHP script.
+
+Configuring Environment Paths
+=============================
+
+The environment variable called :php:`TYPO3_PATH_APP` is used
+to allow to store data outside of the document root.
+
+All composer-based installations benefit from this functionality, as data that was previously
+stored and hard-coded within :file:`typo3temp/var/` is now stored within the **project root** folder :file:`var/`.
+
+For non-composer installations (Classic Mode), it is possible to set the environment variable to a folder, usually one level
+upwards than the regular **webroot**. This increases security for any TYPO3 installation as files are not
+publicly accessible (for example via web browser) anymore.
+
+A typical example:
+
+- :php:`TYPO3_PATH_APP` is set to :file:`/var/www/my-project`.
+- The web folder is then set to :php:`TYPO3_PATH_ROOT` :file:`/var/www/my-project/public`.
+
+Non-public files are then put to
+
+- :file:`/var/www/my-project/var/session` (like Maintenance Tool Session files)
+- :file:`/var/www/my-project/var/cache` (Caching Framework data)
+- :file:`/var/www/my-project/var/lock` (Files related to locking)
+- :file:`/var/www/my-project/var/log` (Files related to logging)
+- :file:`/var/www/my-project/var/extensionmanager` (Files related to extension manager data)
+- :file:`/var/www/my-project/var/transient` (Files related to import/export, core updater, FAL)
```

Comment:
I love that you went through the hassle to list concrete examples here. :+1:

Reply:
We both love Benni I guess ;)